rehttp_server_post17_test.go

```diff
@@ -0,0 +1,136 @@
+//go:build go1.7
+// +build go1.7
+
+package rehttp
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestTransport_RoundTripTimeouts(t *testing.T) {
+	attemptCountMu := sync.Mutex{}
+	attemptCount := 0
```

Comment:
I'd simplify this a bit by using an `int32` and the `atomic` package, e.g. https://pkg.go.dev/sync/atomic@go1.7#AddInt32

Reply:
If I use a buffered channel for synchronization (to eliminate the sleep) each of these will already effectively assert 4 attempts (because the test won't move on until 4 channel reads). That could totally eliminate the need for the `attemptCount`. Downside is that if this test ever breaks it'll break by hanging (waiting on a channel read), which won't make the failure mode obvious. Do you have a preference between the two? (eliminating `attemptCount`, its increments, and any assertions on it vs keeping it around but using `atomic` addition so that an assertion can fail with a clearer message _before_ the channel reads?